artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTUtil.java

```diff
@@ -149,16 +154,52 @@ public static void logMessage(MQTTSessionState state, MqttMessage message, boole
-               for (MqttTopicSubscription sub : ((MqttSubscribeMessage) message).payload().topicSubscriptions()) {
-                  log.append("\n\t" + sub.topicName() + " : " + sub.qualityOfService());
-               }
+            switch (message.fixedHeader().messageType()) {
+               case PUBLISH:
+                  MqttPublishVariableHeader publishHeader = (MqttPublishVariableHeader) message.variableHeader();
+                  log.append("(" + publishHeader.packetId() + ")")
+                     .append(" topic=" + publishHeader.topicName())
+                     .append(", qos=" + message.fixedHeader().qosLevel())
+                     .append(", retain=" + message.fixedHeader().isRetain())
+                     .append(", dup=" + message.fixedHeader().isDup());
+                     //.append(" payload=" + ((MqttPublishMessage)message).payload().toString(StandardCharsets.UTF_8));
+                  break;
+               case CONNECT:
+                  MqttConnectVariableHeader connectHeader = (MqttConnectVariableHeader) message.variableHeader();
+                  MqttConnectPayload payload = ((MqttConnectMessage)message).payload();
+                  log.append(" name=").append(connectHeader.name())
+                     .append(", version=").append(connectHeader.version())
+                     .append(", hasUserName=").append(connectHeader.hasUserName())
+                     .append(", hasPassword=").append(connectHeader.hasPassword())
+                     .append(", isWillRetain=").append(connectHeader.isWillRetain())
+                     .append(", isWillFlag=").append(connectHeader.isWillFlag())
+                     .append(", isCleanSession=").append(connectHeader.isCleanSession())
+                     .append(", keepAliveTimeSeconds=").append(connectHeader.keepAliveTimeSeconds())
+                     .append(", clientIdentifier=").append(payload.clientIdentifier())
+                     .append(", willTopic=").append(payload.willTopic())
+                     .append(", willMessage=").append(payload.willMessage())
+                     .append(", userName=").append(payload.userName());
+                     //.append(", password=").append(payload.password());
+                  break;
+               case SUBSCRIBE:
+                  for (MqttTopicSubscription sub : ((MqttSubscribeMessage) message).payload().topicSubscriptions()) {
+                     log.append("\n\t" + sub.topicName() + " : " + sub.qualityOfService());
+                  }
+                  break;
+               case SUBACK:
+                  for (Integer qos : ((MqttSubAckMessage) message).payload().grantedQoSLevels()) {
+                     log.append("\n\t" + qos);
+                  }
+                  break;
+               case UNSUBSCRIBE:
+                  for (String topic : ((MqttUnsubscribeMessage) message).payload().topics()) {
+                     log.append("\n\t" + topic);
```

Comment:
Ditto as above

Reply:
Packet-ID is already logged above before the switch statement starts.

Reply:
As above you are right sorry !